aten/src/ATen/Context.cpp

```diff
@@ -343,6 +343,40 @@ void Context::setBlasPreferredBackend(at::BlasBackend b) {
 #endif
 }
 
+at::ROCmFABackend Context::getROCmFAPreferredBackend() const {
+  return rocm_fa_preferred_backend;
+}
+
+void Context::setROCmFAPreferredBackend(at::ROCmFABackend b) {
+
+  // TODO: add plumbing for hasCK for validity checking
+  TORCH_CHECK((b != at::ROCmFABackend::Ck) || hasROCM(),
+      "Cannot set preferred flash attention backend to Ck if PyTorch has not been compiled for ROCm.");
+#ifdef USE_ROCM
+  if(b == at::ROCmFABackend::Ck) {
+    static const bool ck_unsupported = []() {
+      static const std::vector<std::string> archs = {
+          "gfx90a",  "gfx942"
```

Comment:
Doesn't CK support gfx908 too and gfx11xx too?

Reply:
No it doesnt support it, currently this is the support matrix.

Reply:
Ah I see. CK supports those, but the Flash Attention CK backend does not support them.

The upstream https://github.com/Dao-AILab/flash-attention library does support these gfx but it's via triton and only on with `FLASH_ATTENTION_TRITON_AMD_ENABLE`, not via the files you added here.

It's very confusing keeping track of all these support matrices :sweat_smile: